reconcile/utils/external_resource_spec.py

```diff
@@ -113,7 +113,15 @@ def cluster_name(self) -> str:
 
     @property
     def output_prefix(self) -> str:
-        return f"{self.identifier}-{self.provider}"
+        # Adhere to DNS-1123 subdomain names spec. It's reasonable to have provider
+        # names that have underscores, but without replacing them with hyphens we run
+        # into issues. Alternatively, we could change Cloudflare worker_script to
+        # worker-script and prevent the use of underscores going forward.
+        #
+        # More info can be found at:
+        # https://kubernetes.io/docs/concepts/overview/working-with-objects/names/.
+        provider = self.provider.replace("_", "-")
+        return f"{self.identifier}-{provider}"
```

Comment:
@geoberle wouldn't mind a second set of eyes here to make sure this isn't going to cause problems. We've never enforced that providers needed to use hyphens, and as a result `worker_script` in Cloudflare ended up with an underscore. I'd suspect if there were going to be an issues it'd be that output names changed and we'd see a diff when running terraform-resources, which I don't.

The alternative would be to change `worker_script` to `worker-script`. In theory, not a huge deal, but it is a breaking change and something we'd need to update in a few places. If we were going to do that I think it'd also make sense to add some enforcement related to provider naming.

Reply:
Thanks for the sanity check. As Gerd also pointed out, if anything was using an underscore it would already be failing `oc apply`. I also compared the tf.json outputs of `terraform-resources` and `terraform-cloudflare-resources` and there aren't any changes.